data_migration.go

```diff
@@ -0,0 +1,53 @@
+package influxdb
+
+import (
+	"context"
+	"time"
+)
+
+// DataMigrationService will transfer the old stored data to new released version.
+type DataMigrationService interface {
+	// CheckAndMigrate will determine if data already migrated.
+	// Usually we will put an index to make sure this the newest version.
+	// Then do a scan to the storage and convert every thing related.
+	CheckAndMigrate(ctx context.Context) error
+	BucketMigrationService
+}
+
+// BucketIsMigratedIndex will be the index check to determine
+// if the newest bucket schema is applied.
+var BucketIsMigratedIndex = []byte("bucketIsMigrated_org")
```

Comment:
Why is this defined here?

Reply:
Because the reason above. This is should be changed any time, there is a schema change. So it should not global for all storage service.